packages/abap-deploy-config-inquirer/src/prompts/questions/abap-target.ts

```diff
@@ -158,18 +158,36 @@ function getUrlPrompt(
  * @param backendTarget - backend target
  * @returns confirm question for scp
  */
-function getScpPrompt(backendTarget?: BackendTarget): Question<AbapDeployConfigAnswersInternal> {
-    return {
-        when: (previousAnswers: AbapDeployConfigAnswersInternal): boolean => showScpQuestion(previousAnswers),
-        type: 'confirm',
-        name: abapDeployConfigInternalPromptNames.scp,
-        message: t('prompts.target.scp.message'),
-        guiOptions: {
-            breadcrumb: t('prompts.target.scp.breadcrumb')
+function getScpPrompt(backendTarget?: BackendTarget): Question<AbapDeployConfigAnswersInternal>[] {
+    const prompts: (ConfirmQuestion<AbapDeployConfigAnswersInternal> | Question)[] = [
+        {
+            when: (previousAnswers: AbapDeployConfigAnswersInternal): boolean => showScpQuestion(previousAnswers),
+            type: 'confirm',
+            name: abapDeployConfigInternalPromptNames.scp,
+            message: t('prompts.target.scp.message'),
+            guiOptions: {
+                breadcrumb: t('prompts.target.scp.breadcrumb')
+            },
+            default: (): boolean | undefined => backendTarget?.abapTarget?.scp
+        }
+    ];
+    prompts.push({
+        when: (answers: AbapDeployConfigAnswersInternal): boolean => {
+            const scpChoice = answers[abapDeployConfigInternalPromptNames.scp];
+            const targetChoice = answers[abapDeployConfigInternalPromptNames.targetSystem];
+            // scpChoice by default is true so only update state if target system is a URL
+            if (scpChoice && targetChoice === TargetSystemType.Url) {
```

Comment:
Do you need this condition, my understanding is that the BTP (SCP) confirm prompt is only ever asked when the target system type is url so its a redundant check?

Reply:
Yes @IainSAP as stated in the PR description, this is to cover both CLI and YUI flows. 

Yes, the SCP prompt is only shown when the target URL option is selected. 

However, based on YUI testing, the state wasn't being updated to reflect the user toggling between Yes | No not did it handle when the user changed their choice from a target URL to a target system. This additional prompt ensures all these flows are covered.